### DISCUSSION TITLE
How to extract only Q&A from PDF file

### DISCUSSION THREAD

**Discussion by Pilli68** [pymupdf/PyMuPDF#1770] [General]
Could you please tell me how to extract only Q&A from PDF file . It's look like Q&A book , i don't need header data ,footer data ,logos and other stuff .

**Comment by JorjMcKie** [comment]
As I responded to the mail you sent me:

* You need to say what type of output you require
* Your example PDF is some science text with two-column pages full with formulas and drawings
* If you just want to get rid of header and footer: find top and bottom limits of text on each page and insert the remaining part into a new page of some output PDF. This can be done fairly easily by locating the delimiting top/bottom line drawings.
* Anything beyond this would be an immense effort - if possible at all.

**Reply by Pilli68** [reply]
@ JorjMcKie, Thank you for respond. then what about below marked data. expecting output format is txt.

<img width="877" alt="Screenshot " src="https://user-images.githubusercontent.com/101690590/175228773-be9305a6-b567-4305-a777-6d43e179d1c6.png">

**Reply by JorjMcKie** [reply]
Then extract the text of the page via `page.get_text("dict",flags=fitz.TEXTFLAGS_TEXT,sort=True)`, determine the coordinates of the vertical line that separates left from right page half.
Loop though extraction output determining which page half for each text span.
If left half, skip until encountering "Answer:".

Note that text output will omit all of your nice formulas like the one shown in your picture.

**Reply by Pilli68** [reply]
@JorjMcKie , great for spending your valuable time and reply for each and every post.  
Could you please guide me /**send me a python script** for removing **only** header, footer data from pdf file (NOTE:  each file in pdf contains different header/footer size and font size. so i am unable to segregate that one .

**Reply by JorjMcKie** [reply]
Sorry, but I cannot, because I am very busy with paid work - which also has a deadline.

And even if that were not the case, your request is a lot beyond general support, especially also regarding the fact that numerous example scripts exist in the "Utilities" sister repo, that can help you doing this.
PyMuPDF admittedly is a large and complex package, so you must invest time to learn using it.

> As a side note: PDF knows nothing about headers and footers - it is always just regular text. You must use any background knowledge you may have of the type of file and then your own wits for how to identify headers / footers. There is no way to write a general recipe for this.
